### PR TITLE
Add wasm64 support.

### PR DESCRIPTION
**Pull request by freeformstu**
This PR adds support for `wasm64-unknown-unknown` although there are no prebuilt `wasm64-unknown-unknown` rustc toolchains. Compiling for `wasm64` will be left up to anyone using this target triple until we get T2 support for wasm64.

### CODE REVIEW

File: rust/platform/platform.bzl [hunk 1]
```diff
@@ -67,13 +67,21 @@ def declare_config_settings():
         )
 
     native.platform(
-        name = "wasm",
+        name = "wasm32",
```
Comment (UebelAndre) [inline rust/platform/platform.bzl:70]: Can you add an alias for backward compatibility?

File: rust/private/dummy_cc_toolchain/BUILD.bazel [hunk 2]
```diff
@@ -1,20 +1,30 @@
+"""When compiling Rust code for wasm (wasm32 or wasm64), we avoid linking to cpp code so we introduce a dummy cc
+toolchain since we know we'll never look it up.
+
+TODO(jedmonds@spotify.com): Need to support linking C code to rust code when compiling for wasm32 or wasm64.
```
Comment (UebelAndre) [inline rust/private/dummy_cc_toolchain/BUILD.bazel:4]: What's this TODO and who is this?

**Review by UebelAndre — CHANGES_REQUESTED** [review]

File: rust/private/dummy_cc_toolchain/BUILD.bazel (hunk 2 above)
Comment (freeformstu) [reply rust/private/dummy_cc_toolchain/BUILD.bazel:4]: I moved this up from the comment below.

**Comment by UebelAndre** [comment]
I think the failing windows example is Ok to disable with a reference to:
- https://github.com/bazelbuild/rules_rust/issues/2878

**Review by UebelAndre — CHANGES_REQUESTED** [review]
`wasm64-unknown-unknown` doesn't have a `std` artifact according to https://doc.rust-lang.org/nightly/rustc/platform-support.html#tier-3

This would lead to failures in analysis (failing to fetch an artifact that doesn't exist) should anything match that platform. I think it should be removed as a default platform and users can use `crates_repository.supported_platform_triples` if they were to have a usable toolchain.

**Review by UebelAndre — APPROVED** [review]

**Pull request merged**